File: segundos 4 lenguajes/Java/JugadorMagico.java
package Java;
abstract class JugadorMagico {
    String nombre;
    int nivel;

    public JugadorMagico() {
        nombre = "Mago";
        nivel = 5;
    }

    public JugadorMagico(String nombre) {
        this.nombre = nombre;
        this.nivel = 10;
    }

    public JugadorMagico(String nombre, int nivel) {
        this.nombre = nombre;
        this.nivel = nivel;
    }

    public JugadorMagico(JugadorMagico j) {
        this.nombre = j.nombre;
        this.nivel = j.nivel;
    }

    abstract void lanzarHechizo();
}

File: segundos 4 lenguajes/js/JugadorMagico.js
export class JugadorMagico {
    constructor(nombre = "Mago", nivel = 5, copia = null) {
        if (copia) {
            this.nombre = copia.nombre;
            this.nivel = copia.nivel;
        } else {
            this.nombre = nombre;
            this.nivel = nivel;
        }
    }

    lanzarHechizo() {
        throw new Error("Método lanzarHechizo() debe implementarse.");
    }
}
